src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/6.diff.sql

```diff
@@ -0,0 +1,77 @@
+/*************************************************************
+    Search Parameter Status Information
+**************************************************************/
+
+ALTER TABLE dbo.SearchParam
+ADD
+    Status varchar(10) NULL,
+    LastUpdated datetimeoffset(7) NULL,
+    IsPartiallySupported bit NULL
+
+CREATE TYPE dbo.SearchParamTableType_1 AS TABLE
```

Comment:
>_1 [](start = 36, length = 2)

is it normal have type with `_` in it?

Reply:
We adopted this naming convention for table-valued parameters because they cannot be updated, deleted or inserted into (more info [here](https://docs.microsoft.com/en-us/sql/relational-databases/tables/use-table-valued-parameters-database-engine?view=sql-server-ver15#Restrictions)).

You aren't the first person to ask this question (see John's comment [here](https://github.com/microsoft/fhir-server/pull/1169#discussion_r465217000)) - maybe we should consider documenting this somewhere? @johnstairs

Reply:
Ok, this is part of accepted convention.
I still not sure I understand why this convention and not some other, but that's irrelevant to my question :)

---
In reply to: [520081613](https://github.com/microsoft/fhir-server/pull/1430#discussion_r520081613) [](ancestors = 520081613)

Reply:
Noticed @brendankowitz also brought this up [here](https://github.com/microsoft/fhir-server/pull/1191#discussion_r473159762). Would be good to have a short chat on the naming convention for table-valued params.

Reply:
Not sure this really needs a meeting, but I created a note to set up a Teams chat about this tomorrow morning.

Reply:
Added a comment about this in 577f673.